File: tests/js/cohere/test_openllmetry.ts
/**
 * Conformance test: OpenLLMetry (Traceloop) JS Cohere instrumentation.
 */

import { CohereInstrumentation } from "@traceloop/instrumentation-cohere";
import { run } from "./common";

function instrument(cohereModule: any) {
  const instrumentation = new CohereInstrumentation();
  instrumentation.manuallyInstrument(cohereModule);
}

run("OpenLLMetry JS: Cohere Conformance Test", instrument).catch((e) => { console.error(e); process.exit(1); });

File: tests/python/aws-bedrock-agent/test_openllmetry.py
"""Conformance test: OpenLLMetry (Traceloop) Bedrock instrumentation for Agent Runtime.

Exercises: invoke_agent
against a mock Bedrock Agent server, with the Traceloop Bedrock instrumentation.
"""

from common import run, run_invoke_agent


def instrument():
    from opentelemetry.instrumentation.bedrock import BedrockInstrumentor
    BedrockInstrumentor().instrument()


if __name__ == "__main__":
    run(
        "OpenLLMetry: AWS Bedrock Agent Conformance Test",
        instrument,
        [run_invoke_agent],
    )
